### PR TITLE
Feature/tao 4696 compile pci

### PR DESCRIPTION
**Pull request by ssipasseuth**
Added the grunt task for portable element compilation.
Edit : the compilation task for portable element has been moved to the PR of taoQtiItem extension: https://github.com/oat-sa/extension-tao-itemqti/pull/986.
It only remains some global configuration required to run that other task properly from the other extension.

### CODE REVIEW

File: views/build/grunt/portableelement.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+ * @example compile all portable element in the extension qtiItemPci
+ * grunt portableelement -e=qtiItemPci
+ *
+ * @example compile only the likertScaleInteraction in the extension qtiItemPci
+ * grunt portableelement -e=qtiItemPci -i=likertScaleInteraction
+ */
+module.exports = function (grunt) {
+    'use strict';
+
+    var Promise = require('pinkie-promise');
```
Comment (krampstudio) [inline views/build/grunt/portableelement.js]: node.js has native Promise support from a while back ([> 0.12.8](http://node.green/#ES2015-built-ins-Promise). What we can do instead is to require it in the package.json : 

```
"engines" : { "node" : ">=0.12.8" }
```

File: views/build/grunt/portableelement.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
+/**
+ * Register the portable element compilation task for a given TAO extension
+ *
+ * @example compile all portable element in the extension qtiItemPci
+ * grunt portableelement -e=qtiItemPci
+ *
+ * @example compile only the likertScaleInteraction in the extension qtiItemPci
+ * grunt portableelement -e=qtiItemPci -i=likertScaleInteraction
+ */
+module.exports = function (grunt) {
```
Comment (krampstudio) [inline views/build/grunt/portableelement.js]: You should be able to register this task from the taoQtiItem extension.

File: views/build/grunt/portableelement.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            }
+        });
+        return model;
+    }
+
+    grunt.registerTask('portableelement', 'Compile Portable Elements', function () {
+
+        var done = this.async();//async mode because requirejs optimization is an async process
+        var extension = grunt.option('e');
+        var selectedId = grunt.option('i');
```
Comment (krampstudio) [inline views/build/grunt/portableelement.js]: You could also support the long version for the parameter (like in other tasks)  in addition to the shorthand : 

```
--extension=qtiItemPci 
--interaction=likerScale
```

File: views/build/grunt/portableelement.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                if(selectedId && selectedId !== model.id){
+                    //not the targeted one
+                    return resolve(report);
+                }
+
+                report.push(grunt.log.subhead.bind(null, model.type + ' "' + model.id + '" found in manifest "' + file + '" ...'));
+
+                if (!model.runtimeHook) {
+                    //when no source file has been found, skip the compilation
+                    report.push(grunt.log.ok.bind(null, 'No source file for ' + model.type +' "' + model.id + '"'));
```
Comment (krampstudio) [inline views/build/grunt/portableelement.js]: Why are you using the `.bind` method ?

Comment (ssipasseuth) [inline views/build/grunt/portableelement.js]: It is for the messages to stay grouped and ordered while each async optimization action run.
With a direct call, the messages ordered are messed up and we can get one message from one compilation task under another one...

File: views/build/grunt/portableelement.js (hunk 3 above)
Comment (ssipasseuth) [inline views/build/grunt/portableelement.js]: ok

File: views/build/grunt/portableelement.js (hunk 2 above)
Comment (ssipasseuth) [inline views/build/grunt/portableelement.js]: ok, I will move it to taoQtiItem

**Comment by ssipasseuth** [comment]
OK, so the grunt task has been moved to the taoQtiItem https://github.com/oat-sa/extension-tao-itemqti/pull/986
Please take a look again when you have some time. Thanks.

**Comment by krampstudio** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request merged**